planner/core/preprocess.go

```diff
@@ -184,6 +188,8 @@ type preprocessor struct {
 	tableAliasInJoin []map[string]interface{}
 	withName         map[string]interface{}
 
+	staleReadProcessor staleread.Processor
```

Comment:
Though I'm OK to make this change, does this mean that we may have more and more `xxxProcessor` fields in the future? Do we have any idea on how to simplify this part?

Reply:
Having a `staleReadProcessor` here is to decouple the stale read code from preprocessor and expect to reuse it. It is only a choice for stale read and I think for other scenes the design will not be limited to 'xxxProcessor'.

For stale read, in the future, there is still only one processor to process stale read. However, we can have multiple subclasses inherits `staleread.Processor` and we can use one of them according to the exact scenario